src/nbl/video/COpenGLShader.h

```diff
@@ -16,10 +16,11 @@ namespace video
 class COpenGLShader : public IGPUShader
 {
 	public:
-		COpenGLShader(core::smart_refctd_ptr<asset::ICPUBuffer>&& _spirv) : m_code(std::move(_spirv)), m_containsGLSL(false) {}
-		COpenGLShader(core::smart_refctd_ptr<asset::ICPUBuffer>&& _glsl, buffer_contains_glsl_t buffer_contains_glsl) : m_code(std::move(_glsl)), m_containsGLSL(true) {}
+		COpenGLShader(ILogicalDevice* dev, core::smart_refctd_ptr<asset::ICPUBuffer>&& _spirv) : IGPUShader(dev), m_code(std::move(_spirv)), m_containsGLSL(false) {}
+		COpenGLShader(ILogicalDevice* dev, core::smart_refctd_ptr<asset::ICPUBuffer>&& _glsl, buffer_contains_glsl_t buffer_contains_glsl) : IGPUShader(dev), m_code(std::move(_glsl)), m_containsGLSL(true) {}
 
 		const asset::ICPUBuffer* getSPVorGLSL() const { return m_code.get(); };
+		const core::smart_refctd_ptr<asset::ICPUBuffer>& getSPVorGLSL_refctd() const { return m_code; };
```

Comment:
what is this method needed for?

Reply:
no idea, i dont think its my code

Reply:
well its definitely not on master

Reply:
probably cyprian's gcc/clang build fixes